mmv1/products/dlp/api.yaml

```diff
@@ -1602,3 +1602,713 @@ objects:
+
+                                1.  there is no record present when transforming a given value or
+                                2.  the field is not present when transforming a given value,
+
+                                a default tweak will be used.
+
+                                Note that case (1) is expected when an `InfoTypeTransformation` is applied to both structured and non-structured `ContentItem`s. Currently, the referenced field may be of value type integer or string.
+
+                                The tweak is constructed as a sequence of bytes in big endian byte order such that:
+
+                                *   a 64 bit integer is encoded followed by a single byte of value 1
+                                *   a string is encoded in UTF-8 format followed by a single byte of value 2
+                              properties:
+                                - !ruby/object:Api::Type::String
+                                  name: 'name'
+                                  description: |
+                                    Name describing the field.
+                            - !ruby/object:Api::Type::NestedObject
+                              name: 'surrogateInfoType'
+                              description: |
+                                The custom infoType to annotate the surrogate with. This annotation will be applied to the surrogate by prefixing it with the name of the custom infoType followed by the number of characters comprising the surrogate. The following scheme defines the format: info\_type\_name(surrogate\_character\_count):surrogate
+
+                                For example, if the name of custom infoType is 'MY\_TOKEN\_INFO\_TYPE' and the surrogate is 'abc', the full replacement value will be: 'MY\_TOKEN\_INFO\_TYPE(3):abc'
+
+                                This annotation identifies the surrogate when inspecting content using the custom infoType [`SurrogateType`](https://cloud.google.com/dlp/docs/reference/rest/v2/InspectConfig#surrogatetype). This facilitates reversal of the surrogate when it occurs in free text.
+
+                                In order for inspection to work properly, the name of this infoType must not occur naturally anywhere in your data; otherwise, inspection may find a surrogate that does not correspond to an actual identifier. Therefore, choose your custom infoType name carefully after considering what your data looks like. One way to select a name that has a high chance of yielding reliable detection is to include one or more unicode characters that are highly improbable to exist in your data. For example, assuming your data is entered from a regular ASCII keyboard, the symbol with the hex code point 29DD might be used like so: ⧝MY\_TOKEN\_TYPE
+                              properties:
+                                - !ruby/object:Api::Type::String
+                                  name: 'name'
```

Comment:
it looks like there's also a `version` field as part of the InfoType; do we care about that for this PR?

Reply:
I've added this in [1533163](https://github.com/GoogleCloudPlatform/magic-modules/pull/6886/commits/15331630d785c5539199f4590dfa905bcbc2e4d1) - I think the inconsistency here was because I was copy/pasting reused fields and the version field is a more recent addition to the API